### PR TITLE
convert Gaussian's grads from input oriantion to standard orientation

### LINKED ISSUE #924: The orientation of Gaussian grads is differ from that of Gaussian atomcoords.
For `grads` parsed data for  Gaussian, input orientation is used,  although standard orientation is used for other data including `atomcoods` and `vibdisps`. This is becauses that Gaussian outputs gradient with input orientation and other values with standard orientatoin.
Since parsed data for Gaussian doesn't contain atom coordinates with input oriea or rotation matrix to convert input orientation to standard orientation, we can't convert parsed `grads` from input orientation to standard orientation.

I think there are two choices to resolve this problem.
1. Convert `grads` to standared orientation inside cclib
2. Add input orietation atom coordinates or rotation matrix to parsed data, so that user can convert `grads` from input orientation to standard  orientation.

Which option do you prefer?
I think 1. is better choice althogh it does'nt maintain backwaord compatibiilty.

I put `atomcoords` and `grads` pased from test data with cclib. You can find that orientation for `grads` is differ from that for `atomcoords`.

``` 
In [2]:
d = cclib.io.ccread("../data/Gaussian/basicGaussian09/dvb_gopt.out")
print(d.atomcoords[0])
[[ 1.205666e+00  6.768910e-01  0.000000e+00]
 [ 2.200000e-03  1.375535e+00  0.000000e+00]
 [-1.205666e+00  6.892860e-01  0.000000e+00]
 [-1.205666e+00 -6.768910e-01  0.000000e+00]
 [-2.200000e-03 -1.375535e+00  0.000000e+00]
 [ 6.730000e-03  2.470443e+00  0.000000e+00]
 [-2.150459e+00  1.244759e+00  0.000000e+00]
 [-6.730000e-03 -2.470443e+00  0.000000e+00]
 [-2.492798e+00 -1.419218e+00  0.000000e+00]
 [-2.577785e+00 -2.759452e+00  0.000000e+00]
 [-1.654345e+00 -3.363669e+00  0.000000e+00]
 [-3.397785e+00 -7.969190e-01  0.000000e+00]
 [-3.516212e+00 -3.273509e+00  0.000000e+00]
 [ 2.492798e+00  1.419218e+00  0.000000e+00]
 [ 3.397785e+00  7.969190e-01  0.000000e+00]
 [ 2.577785e+00  2.759452e+00  0.000000e+00]
 [ 1.654345e+00  3.363669e+00  0.000000e+00]
 [ 3.516212e+00  3.273509e+00  0.000000e+00]
 [ 1.205666e+00 -6.892860e-01  0.000000e+00]
 [ 2.150459e+00 -1.244759e+00  0.000000e+00]]

In [3]:
print(d.grads[0])
[[-0.05044906  0.          0.0006004 ]
 [-0.00722221  0.          0.0239222 ]
 [-0.03871192  0.          0.02960709]
 [ 0.05044906  0.         -0.0006004 ]
 [ 0.00722221  0.         -0.0239222 ]
 [ 0.00108782  0.          0.00136294]
 [ 0.00068463  0.          0.00135426]
 [-0.00108782  0.         -0.00136294]
 [ 0.01037156  0.          0.0028839 ]
 [-0.01923258  0.         -0.01367979]
 [ 0.00632887  0.          0.00841993]
 [ 0.00120737  0.          0.00128327]
 [ 0.01947715  0.          0.0103679 ]
 [-0.01037156  0.         -0.0028839 ]
 [-0.00120737  0.         -0.00128327]
 [ 0.01923258  0.          0.01367979]
 [-0.00632887  0.         -0.00841993]
 [-0.01947715  0.         -0.0103679 ]
 [ 0.03871192  0.         -0.02960709]
 [-0.00068463  0.         -0.00135426]] ```

### PR DESCRIPTION
**Pull request by cks-coil**
To resolve #924 , I add codes to get rotations for two different orientations and convert Gaussian's `grads` from input orientation to standard orientation. I also add tests for them.

I put parse data with modified codes.
```In [2]:
d = cclib.io.ccread("../data/Gaussian/basicGaussian09/dvb_gopt.out")
print(d.atomcoords[0])
[[ 1.205666e+00  6.768910e-01  0.000000e+00]
 [ 2.200000e-03  1.375535e+00  0.000000e+00]
 [-1.205666e+00  6.892860e-01  0.000000e+00]
 [-1.205666e+00 -6.768910e-01  0.000000e+00]
 [-2.200000e-03 -1.375535e+00  0.000000e+00]
 [ 6.730000e-03  2.470443e+00  0.000000e+00]
 [-2.150459e+00  1.244759e+00  0.000000e+00]
 [-6.730000e-03 -2.470443e+00  0.000000e+00]
 [-2.492798e+00 -1.419218e+00  0.000000e+00]
 [-2.577785e+00 -2.759452e+00  0.000000e+00]
 [-1.654345e+00 -3.363669e+00  0.000000e+00]
 [-3.397785e+00 -7.969190e-01  0.000000e+00]
 [-3.516212e+00 -3.273509e+00  0.000000e+00]
 [ 2.492798e+00  1.419218e+00  0.000000e+00]
 [ 3.397785e+00  7.969190e-01  0.000000e+00]
 [ 2.577785e+00  2.759452e+00  0.000000e+00]
 [ 1.654345e+00  3.363669e+00  0.000000e+00]
 [ 3.516212e+00  3.273509e+00  0.000000e+00]
 [ 1.205666e+00 -6.892860e-01  0.000000e+00]
 [ 2.150459e+00 -1.244759e+00  0.000000e+00]]


In [3]:
print(d.grads[0])
[[ 2.48091490e-02  4.39314689e-02  2.77548848e-18]
 [-1.70627408e-02  1.82563631e-02 -5.95048205e-19]
 [-6.16954350e-03  4.83438670e-02  9.16300244e-19]
 [-2.48091490e-02 -4.39314689e-02 -2.77548848e-18]
 [ 1.70627408e-02 -1.82563631e-02  5.95048205e-19]
 [-1.72486383e-03 -2.56511599e-04 -1.17129917e-19]
 [-1.51493650e-03  8.78288353e-05 -9.43871231e-20]
 [ 1.72486383e-03  2.56511599e-04  1.17129917e-19]
 [-7.70123000e-03 -7.52178374e-03 -6.95803873e-19]
 [ 2.14866236e-02  9.76491600e-03  1.63715802e-18]
 [-1.04593019e-02 -1.24611701e-03 -7.01872810e-19]
 [-1.71598614e-03 -3.99897730e-04 -1.20449400e-19]
 [-1.87451833e-02 -1.16391976e-02 -1.51284969e-18]
 [ 7.70123000e-03  7.52178374e-03  6.95803873e-19]
 [ 1.71598614e-03  3.99897730e-04  1.20449400e-19]
 [-2.14866236e-02 -9.76491600e-03 -1.63715802e-18]
 [ 1.04593019e-02  1.24611701e-03  7.01872810e-19]
 [ 1.87451833e-02  1.16391976e-02  1.51284969e-18]
 [ 6.16954350e-03 -4.83438670e-02 -9.16300244e-19]
 [ 1.51493650e-03 -8.78288353e-05  9.43871231e-20]]
```

### CODE REVIEW

**Review by berquist — CHANGES_REQUESTED** [review]
This looks good to me, just waiting for the CI fix and someone else's eyes.

**Comment by shivupa** [comment]
I agree with the proposed changes and this looks good to me. I think the testing is adequate. This sounds like an appropriate solution to #924. If we find another code that does this then it can probably be handled in the same way. I am going to approve the PR

**Review by shivupa — APPROVED** [review]

**Comment by berquist** [comment]
I wonder if we should put this behind `--future` after all and have the next release be 1.7? @langner

**Comment by langner** [comment]
As discussed today, we want to fix this, since it's really a bug. It's OK to be backwards incompatible in that case.

@cks-coil Could you perhaps add an entry to the [data notes](https://github.com/cclib/cclib/blob/master/doc/sphinx/data_notes.rst) about `grads` and explicitly mention we want to guaranty the grads will be aligned with atomcoords?

**Review by shivupa — APPROVED** [review]

File: test/data/testGeoOpt.py [hunk 1]
```diff
@@ -246,6 +246,14 @@ def testoptdone(self):
         self.assertTrue(sum(convergence) >= 2)
 
 
+class GaussianGeoOptTest(GenericGeoOptTest):
+    """Customzed geometry optimization unittest"""
```
Comment (shivupa) [inline test/data/testGeoOpt.py:250]: Typo

File: cclib/parser/utils.py [hunk 2]
```diff
@@ -155,6 +156,54 @@ def convertor(value, fromunits, tounits):
...
+    if hasattr(scipy.spatial.transform.Rotation, "align_vectors"):
+        r, _ = scipy.spatial.transform.Rotation.align_vectors(b_, a_)
+    else:
+        if numpy.linalg.matrix_rank(a_) == 1:
+            # in the case of linear molecule, e.g. O2, C2H2
+            idx = numpy.argmax(numpy.linalg.norm(a_, ord=2, axis=1))
+            rmat = _get_rmat_from_vecs(a_[idx], b_[idx])
+            r = scipy.spatial.transform.Rotation.from_dcm(rmat)
+        else:
+            # we need to remove b_[0] and a_[1] ( both of them are [0,0,0] ) to avoid SVD unconvergence error
```
Comment (shivupa) [inline cclib/parser/utils.py:204]: ```# we need to remove b_[0] and a_[0***]```

**Review by shivupa — CHANGES_REQUESTED** [review]

**Comment by shivupa** [comment]
Sorry I had approved thinking the travis failure would be an easy fix I am confused as to why this is failing

**Comment by shivupa** [comment]
Test
```
import cclib

data = cclib.io.ccread("~/Documents/cclib/data/Gaussian/basicGaussian09/dvb_gopt.out")
print(data.grads[:,:,2])
print("-"*80)
data = cclib.io.ccread("~/Documents/cclib/data/Gaussian/basicGaussian16/dvb_gopt.out")
print(data.grads[:,:,2])
```

python 3 output:
```
[[ 5.55097696e-18 -1.19009641e-18  1.83260049e-18 -5.55097696e-18
   1.19009641e-18 -2.34259834e-19 -1.88774246e-19  2.34259834e-19
  -1.39160775e-18  3.27431604e-18 -1.40374562e-18 -2.40898801e-19
  -3.02569939e-18  1.39160775e-18  2.40898801e-19 -3.27431604e-18
   1.40374562e-18  3.02569939e-18 -1.83260049e-18  1.88774246e-19]
 [ 6.62932764e-19  5.55811674e-19  1.39937378e-18 -6.62932764e-19
  -5.55811674e-19 -6.86535273e-20 -2.20718055e-19  6.86535273e-20
   1.00845349e-18 -1.01277653e-19  8.99476105e-19 -1.38086931e-19
   4.06285616e-19 -1.00845349e-18  1.38086931e-19  1.01277653e-19
  -8.99476105e-19 -4.06285616e-19 -1.39937378e-18  2.20718055e-19]
 [ 4.04312028e-19 -3.13999993e-19  4.26375713e-19 -4.04312028e-19
   3.13999993e-19  4.90101293e-20 -4.97355490e-20 -4.90101293e-20
   2.04359307e-19  5.69683745e-20 -2.30712449e-19 -1.38911105e-20
   5.42614842e-20 -2.04359307e-19  1.38911105e-20 -5.69683745e-20
   2.30712449e-19 -5.42614842e-20 -4.26375713e-19  4.97355490e-20]
 [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
 [-8.71547279e-21 -9.37583344e-23  5.82478510e-22  8.71547279e-21
   9.37583344e-23 -7.28028748e-22  1.95843342e-22  7.28028748e-22
  -8.94517793e-21  1.72528658e-22  2.71072054e-21  1.26526567e-21
   9.30644450e-22  8.94517793e-21 -1.26526567e-21 -1.72528658e-22
  -2.71072054e-21 -9.30644450e-22 -5.82478510e-22 -1.95843342e-22]]
--------------------------------------------------------------------------------
[[ 5.55292806e-18 -1.18854285e-18  1.82485807e-18 -5.55292806e-18
   1.18854285e-18 -2.38535053e-19 -1.88598887e-19  2.38535053e-19
  -1.38993919e-18  3.27271452e-18 -1.40761861e-18 -2.40371667e-19
  -3.02401010e-18  1.38993919e-18  2.40371667e-19 -3.27271452e-18
   1.40761861e-18  3.02401010e-18 -1.82485807e-18  1.88598887e-19]
 [ 1.14904641e-19  3.71472408e-19  4.65185446e-19 -1.14904641e-19
  -3.71472408e-19 -8.69401218e-20 -7.37921946e-20  8.69401218e-20
   2.83477575e-19 -1.40560452e-19  3.93530764e-19 -7.63714647e-20
   1.16462173e-19 -2.83477575e-19  7.63714647e-20  1.40560452e-19
  -3.93530764e-19 -1.16462173e-19 -4.65185446e-19  7.37921946e-20]
 [-3.24796079e-19  3.55979468e-19 -3.45540707e-19  3.24796079e-19
  -3.55979468e-19  3.26804694e-20  9.88552573e-20 -3.26804694e-20
  -4.08139300e-19  9.28424004e-20  1.17103660e-19  1.04411424e-19
   1.53591584e-20  4.08139300e-19 -1.04411424e-19 -9.28424004e-20
  -1.17103660e-19 -1.53591584e-20  3.45540707e-19 -9.88552573e-20]
 [ 5.78153081e-20 -7.49156293e-20 -2.97117886e-21 -5.78153081e-20
   7.49156293e-20 -1.12168053e-20 -5.17719201e-21  1.12168053e-20
   5.74547077e-20 -4.62816452e-20  2.70177214e-20 -1.84643412e-20
  -3.18034488e-21 -5.74547077e-20  1.84643412e-20  4.62816452e-20
  -2.70177214e-20  3.18034488e-21  2.97117886e-21  5.17719201e-21]
 [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]]
```

python2 output:
```
[[-4.39314689e-02 -1.82563631e-02 -4.83438670e-02  4.39314689e-02
   1.82563631e-02  2.56511602e-04 -8.78288329e-05 -2.56511602e-04
   7.52178375e-03 -9.76491603e-03  1.24611702e-03  3.99897733e-04
   1.16391977e-02 -7.52178375e-03 -3.99897733e-04  9.76491603e-03
  -1.24611702e-03 -1.16391977e-02  4.83438670e-02  8.78288329e-05]
 [ 9.35945488e-19 -1.78121234e-18 -2.32863728e-21 -9.35945488e-19
   1.78121234e-18  6.03651795e-19  1.74582016e-20 -6.03651795e-19
   4.88450780e-19  1.03456643e-18 -8.56595961e-19  2.34483266e-19
   1.39058598e-19 -4.88450780e-19 -2.34483266e-19 -1.03456643e-18
   8.56595961e-19 -1.39058598e-19  2.32863728e-21 -1.74582016e-20]
 [-6.74017286e-19  4.62194949e-19 -7.07921721e-19  6.74017286e-19
  -4.62194949e-19 -1.26243460e-19  4.65210093e-20  1.26243460e-19
  -1.88736582e-19 -1.80113480e-19  4.26074731e-19 -3.38569173e-20
  -1.25554234e-19  1.88736582e-19  3.38569173e-20  1.80113480e-19
  -4.26074731e-19  1.25554234e-19  7.07921721e-19 -4.65210093e-20]
 [ 4.45826709e-21  1.16080479e-20  2.02112216e-20 -4.45826709e-21
  -1.16080479e-20  5.56366064e-21  2.64277489e-21 -5.56366064e-21
  -8.80756579e-21  2.37841413e-20 -1.13130061e-20 -3.72868403e-22
  -2.78344015e-21  8.80756579e-21  3.72868403e-22 -2.37841413e-20
   1.13130061e-20  2.78344015e-21 -2.02112216e-20 -2.64277489e-21]
 [ 1.45843337e-20  5.19118082e-21 -3.47299967e-21 -1.45843337e-20
  -5.19118082e-21  7.81819054e-22  8.94395669e-22 -7.81819054e-22
   1.73945303e-20 -1.94377847e-21 -4.93960428e-21 -1.26942901e-21
  -5.38902256e-22 -1.73945303e-20  1.26942901e-21  1.94377847e-21
   4.93960428e-21  5.38902256e-22  3.47299967e-21 -8.94395669e-22]]
--------------------------------------------------------------------------------
[[-4.39284775e-02 -1.82225584e-02 -4.83155608e-02  4.39284775e-02
   1.82225584e-02  2.38414490e-04 -8.79682082e-05 -2.38414490e-04
   7.52866538e-03 -9.80614416e-03  1.27417732e-03  4.12945900e-04
   1.16339840e-02 -7.52866538e-03 -4.12945900e-04  9.80614416e-03
  -1.27417732e-03 -1.16339840e-02  4.83155608e-02  8.79682082e-05]
 [-8.64774252e-19  4.27071156e-19 -9.40306943e-19  8.64774252e-19
  -4.27071156e-19 -2.23553287e-19  1.35190859e-19  2.23553287e-19
  -8.90750029e-19 -4.01766842e-19 -2.19703367e-19 -5.56443780e-21
  -3.23783556e-19  8.90750029e-19  5.56443780e-21  4.01766842e-19
   2.19703367e-19  3.23783556e-19  9.40306943e-19 -1.35190859e-19]
 [ 4.44564496e-19 -4.58796112e-19  4.71272577e-19 -4.44564496e-19
   4.58796112e-19 -2.31547004e-20 -1.18116184e-19  2.31547004e-20
   4.86157503e-19 -8.65446048e-20 -1.78866588e-19 -1.15587429e-19
  -4.22717417e-21 -4.86157503e-19  1.15587429e-19  8.65446048e-20
   1.78866588e-19  4.22717417e-21 -4.71272577e-19  1.18116184e-19]
 [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
   0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
 [-9.62746549e-21 -1.92140748e-21  1.86584082e-21  9.62746549e-21
   1.92140748e-21 -1.55209179e-22  5.66213743e-23  1.55209179e-22
  -1.14707688e-20  3.53495011e-22  2.33540964e-21  9.36917210e-22
  -3.32511796e-23  1.14707688e-20 -9.36917210e-22 -3.53495011e-22
  -2.33540964e-21  3.32511796e-23 -1.86584082e-21 -5.66213743e-23]]
```

File: cclib/parser/utils.py [hunk 3]
```diff
@@ -155,6 +156,54 @@ def convertor(value, fromunits, tounits):
...
+    """
+    assert a.shape == b.shape
+    if a.shape[0] == 1:
+        return scipy.spatial.transform.Rotation.from_euler('xyz', [0,0,0])
+    # remove translation part
+    a_ = a - a[0]
+    b_ = b - b[0]
+    if hasattr(scipy.spatial.transform.Rotation, "align_vectors"):
+        r, _ = scipy.spatial.transform.Rotation.align_vectors(b_, a_)
+    else:
```
Comment (shivupa) [inline cclib/parser/utils.py:197]: I think the problem is somewhere in this section

**Review by shivupa — APPROVED** [review]

**Review by berquist — APPROVED** [review]

**Pull request merged**